### PR TITLE
chore: add tests for management canister types in PocketIC

### PR DESCRIPTION
**Pull request by mraszyk**
This PR adds tests for candid equality of management canister types declared in the PocketIC library with [types](https://github.com/dfinity/interface-spec/blob/master/spec/_attachments/ic.did) specified in the official Interface Specification of the Internet Computer.

### CODE REVIEW

**Comment by mraszyk** [comment]
@dsarlis We could consider adding a similar test for the `ic-management-canister-types` crate.

**Comment by dsarlis** [comment]
> @dsarlis We could consider adding a similar test for the `ic-management-canister-types` crate.

I'm not a fan of adding dummy canister endpoints to make the candid equality check work. I'd rather prefer we start producing the types from the candid file when that's stable instead. I believe we have discussed this in the past but we didn't have time to look into it.

**Comment by mraszyk** [comment]
> I'd rather prefer we start producing the types from the candid file when that features is stable instead. I believe we have discussed this in the past but we didn't have time to look into it.

Indeed, but I'm not sure the alternative will be ready soon which is why I went for this approach for PocketIC (better a test that is less ideal than no test in my opinion). Of course, it is up to you to decide how to test ic-management-canister-types.

File: packages/pocket-ic/src/management_canister.rs [hunk 1]
```diff
@@ -26,7 +28,6 @@ pub struct DefiniteCanisterSettings {
     pub wasm_memory_limit: candid::Nat,
     pub memory_allocation: candid::Nat,
     pub compute_allocation: candid::Nat,
-    pub wasm_memory_threshold: candid::Nat,
```
[the remark this answers is not in the document]
Comment (dsarlis) [reply packages/pocket-ic/src/management_canister.rs:29]: Right, I don't have a good idea how to solve that. But indeed it is probably a bigger concern compared to adding some dummy endpoints.

File: MODULE.bazel [hunk 2]
```diff
@@ -835,3 +835,12 @@ http_file(
...
 )
+
+# Management canister candid interface
+
+http_file(
+    name = "management_canister_did",
+    downloaded_file_path = "ic.did",
+    sha256 = "31d4654d60b364420a2e52f546f06b2255dc78ac8c2d768271f004b8946e92cb",
+    url = "https://raw.githubusercontent.com/dfinity/interface-spec/b7cbca243f527b6819a0b0f644f87366a47d3eb4/spec/_attachments/ic.did",
+)
```
Comment (basvandijk) [inline MODULE.bazel:846]: Do we need both this and [`interface_spec` above](https://sourcegraph.com/github.com/dfinity/ic/-/blob/MODULE.bazel?L793-L799)? If so, could you place them side by side to easily spot the differences?

**Review by basvandijk — CHANGES_REQUESTED** [review]

File: MODULE.bazel (hunk 2 above)
Comment (mraszyk) [reply MODULE.bazel]: Thank you for pointing this out! I wasn't aware of the other dependency used in fuzzers so I moved them both together. I'd prefer to not bump it in this PR if you don't mind.

Comment (mraszyk) [reply MODULE.bazel]: @venkkatesh-sekar Should we keep two versions of this external dependency or unify them to the latest master commit in this PR after all?

**Review by basvandijk — APPROVED** [review]

File: MODULE.bazel (hunk 2 above)
Comment (venkkatesh-sekar) [reply MODULE.bazel]: @mraszyk the fuzzers can use the latest master version as well as they are only used for corpus generation.

Comment (mraszyk) [reply MODULE.bazel:838]: Great! I've merged them together into a single dependency pulling the latest version.

**Review by dsarlis — APPROVED** [review]

**Review by michael-weigelt — APPROVED** [review]

**Pull request merged**